### PR TITLE
fix modal in menu setting cannot be closed by pressing ESC

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#15406` modal in menu setting cannot be closed by pressing ESC`](http://www.openproject.org/work_packages/15406)

[`* `#15407` focus lost after closing modal in menu settings`](http://www.openproject.org/work_packages/15407)

I changed a bower.json dependency to my own repository so a bower clean cache and a bower install might be needed

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://0xf013-feature-15406-modal-in-menu-settings-cannot-be-closed-by.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: spec/features/accessibility/work_packages/work_package_query_spec.rb [hunk 1]
```diff
@@ -267,4 +267,38 @@ def click_sort_descending_link
       end
     end
   end
+
+  describe 'settings button', js: true do
```
Comment (myabc) [inline spec/features/accessibility/work_packages/work_package_query_spec.rb:273]: @0xF013 please ignore this warning from @houndci – I don't agree with this convention. I'll try to adjust configuration.

File: spec/features/accessibility/work_packages/work_package_query_spec.rb [hunk 2]
```diff
@@ -267,4 +266,34 @@ def click_sort_descending_link
...
+    before do
+      work_packages_page.visit_index
+    end
```
Comment (0xF013) [inline spec/features/accessibility/work_packages/work_package_query_spec.rb]: @houndci 
bad dog

Comment (myabc) [inline spec/features/accessibility/work_packages/work_package_query_spec.rb]: @0xF013 see #1879

**Comment by ulferts** [comment]
@0xF013 just a pointer that specs are failing.

**Comment by 0xF013** [comment]
@ulferts it's weird, I just ran the spec with the same seed locally and it's green. Also it fails only on one specific config 
https://www.dropbox.com/s/u5mrbs2jkfnsclb/%D0%A1%D0%BA%D1%80%D0%B8%D0%BD%D1%88%D0%BE%D1%82%202014-09-09%2016.13.43.png?dl=0
Do you happend to know who can I ask about this test suite param?

**Comment by ulferts** [comment]
@0xF013 the configurations are splitting up the different kinds of tests we unfortunately still have: specs, cukes, test-unit.
From what I see, the same spec fails on postgres and mysql. So it is a spec that is failing and because it does so on both dbs it is not a flickering test and thus something you will need to fix.

**Comment by ulferts** [comment]
@0xF013 The failing tests on travis probably have to do with a bug we noticed on the work packages page. There is no vertical scrolling defined on the page. So when a small viewport is selected, the settings button is no longer in an accessible area.

So I guess it makes sense to halt working on this until the bug is fixed and see if fixing the bug will also fix those here.

**Comment by ulferts** [comment]
@0xF013 can you please rebase your branch to the current HEAD of dev where the scrolling feature has been added. In https://github.com/opf/openproject/pull/1917 I tested doing that and the test failures did no longer occur.

**Comment by 0xF013** [comment]
@ulferts will do it do today

**Pull request merged**